File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/FindInsteadOfFirstOrDefault.Net.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System

Public Class FindInsteadOfFirstOrDefault
    Public Sub List(data As List(Of Integer))
        Dim unused = data.FirstOrDefault(0) ' Compliant
        unused = data.FirstOrDefault(Function(x) False, 0) ' Noncompliant
        '             ^^^^^^^^^^^^^^
    End Sub

    Public Sub Array(data As Integer())
        Dim unused = data.FirstOrDefault(0) ' Compliant
        unused = data.FirstOrDefault(Function(x) False, 0) ' Noncompliant
        '             ^^^^^^^^^^^^^^
    End Sub
End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/FindInsteadOfFirstOrDefault.Net.cs
using System.Collections.Generic;
using System.Linq;
using System;

public class FindInsteadOfFirstOrDefault
{
    public void List(List<int> data)
    {
        _ = data.FirstOrDefault(default(int)); // Compliant
        _ = data.FirstOrDefault(x => false, default(int)); // Compliant
    }

    public void Array(int[] data)
    {
        _ = data.FirstOrDefault(default(int)); // Compliant
        _ = data.FirstOrDefault(x => false, default(int)); // Compliant
    }
}
